src/System.Management.Automation/engine/Subsystem/Prediction/CommandPrediction.cs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation.Language;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Management.Automation.Subsystem
+{
+    /// <summary>
+    /// The class represents the prediction result from a predictor.
+    /// </summary>
+    public sealed class PredictionResult
+    {
+        /// <summary>
+        /// Gets the Id of the predictor.
+        /// </summary>
+        public Guid Id { get; }
+
+        /// <summary>
+        /// Gets the suggestions.
+        /// </summary>
+        public List<string> Suggestions { get; }
```

Comment:
Do we not want this type to be more similar to `System.Management.Automation.CompletionResult` which contains:

```
CompletionText Property   string CompletionText {get;}
ListItemText   Property   string ListItemText {get;}
ResultType     Property   System.Management.Automation.CompletionResultType ResultType {get;}
ToolTip        Property   string ToolTip {get;}
```

I'd think a ToolTip could be useful for an Azure predictor that explains the Suggestions that comes back.
```pwsh
Get-Az
```
predicts to maybe
```pwsh
Get-AzVM -Name myVM | Start-VM
```
could have tooltip description of "Example of getting a VM and starting it"

Maybe for PSRL it doesn't make sense for prediction text, but for an editor it's really important.

Reply:
Reusing `CompletionResult` or a subclass of it doesn't fit to the prediction results, for example, the `ResultType` and the property name `CompletionText`, also the difference between `ListItemText` and `CompletionText` in prediction may be .. zero, as compared with the more granular and structured completion result, the prediction results should be a complete command line string.

Comparing to tab completion, the prediction API will be called very frequently once it's turned on by user (every char input?), so the GC pressure from creating a lot short-living objects is a big concern, that's why I started with just a list of strings. But I talked to @theJasonHelmick and we will get the feedback from the Aladdin team about whether they want to attach any metadata to each of the suggestion text.

Reply:
I just wanna make sure that PSRL predictions are not the only usecase in mind for `CommandPrediction`.

Reply:
I thought about this some more, and I don't think the frequency matters too much. We run Completions API in PSES at a high frequency and that uses the pipeline. Predictions API will be called just as often as Completions API in editor scenarios.

Reply:
Is completion API triggered at every keystroke in PSES editor, or only at some specific keystrokes like `-`?
GC pressure is a concerns, and that's why it'd better to not generate so many short-living objects. But the main driving force here is whether `tooltip` matters to a prediction result. @theJasonHelmick, I didn't follow up with you on this, did you get any feedback on this from the Alladin team?

Reply:
Theoretically only at specific keystrokes, but VSCode basically gets to decide when it asks for completions. We just try to be reactive.

Reply:
It can do things like ask for completions and then cancel the request, or ask for completions, take some and then renew the request later to ask for more -- filling a completion buffer

Reply:
@TylerLeonhardt @daxian-dbw @rjmholt  I'm having a difficult time in imagining a scenario where a tooltip, along with the existing menu complete and future dynamic help, adds something to the interactive experience.  In recent conversations I've had with Aladdin, they hadn't planned on offering tooltips with their predictions.  I'm not saying this isn't worth considering, just not seeing the benefit at the moment.  Perhaps in vscode it makes more sense?

Reply:
> Perhaps in vscode it makes more sense?

Yes, that's what I said above "Maybe for PSRL it doesn't make sense for prediction text, but for an editor it's really important."

There is no concept of "predictions vs completions" in vscode.... there is just "Intellisense". Intellisense is closer to Menu Complete... but it also has Snippets which are snippets of code that you are _most likely to use in the context_ (but they're not smart). These Aladdin predictions are really just smarter snippets (or at least that's how I view them).

Snippets in vscode are rich... they have descriptions because you wouldn't know what the code is without context.

```json
"Suppress PSScriptAnalyzer Rule on Parameter": {
		"prefix": "suppress-message-rule-for-parameter",
		"description": "Suppress a built-in PSScriptAnalyzer rule on a parameter using the SuppressMessageAttribute",
		"body": [
			"[Diagnostics.CodeAnalysis.SuppressMessageAttribute('${1:PSUseDeclaredVarsMoreThanAssignments}', '${2:ParamName}')]"
		]
```

Reply:
`ToolTip` property added to the suggestion result.